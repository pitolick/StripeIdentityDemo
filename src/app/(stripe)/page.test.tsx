import React from 'react'
import { render } from '@testing-library/react'
import FormStripe, { metadata } from './page'

// HeadingLineをモック化
jest.mock('@/components/elements', () => ({
  HeadingLine: () => <div>HeadingLine</div>,
}))

// FormStripeInputClientをモック化
jest.mock('./_components/FormStripeInputClient', () => ({
  FormStripeInputClient: () => <div>FormStripeInputClient</div>,
}))

// DOMAIN_PRODをモック化
jest.mock('@/config', () => ({
  DOMAIN_PROD: 'example.com',
}))

describe('FormStripeコンポーネント', () => {
  it('正しくレンダリングされること', () => {
    const { container } = render(<FormStripe />)
    expect(container).toMatchSnapshot()
  })

  it('metadataが正しく設定されていること', () => {
    expect(metadata).toHaveProperty('title')
  })
})
