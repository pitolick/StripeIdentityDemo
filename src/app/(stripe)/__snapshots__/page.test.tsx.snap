// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormStripeコンポーネント 正しくレンダリングされること 1`] = `
<div>
  <div
    class="container max-w-5xl pb-10 pt-5 md:pb-32 md:pt-14"
  >
    <div>
      FormStripeInputClient
    </div>
  </div>
</div>
`;
